src/main/java/life/qbic/model/download/OpenbisConnector.java

```diff
@@ -41,15 +63,122 @@ public OpenbisConnector(
+
+    final UploadedDataSetCreation creation = new UploadedDataSetCreation();
+    creation.setUploadId(uploadId);
+    creation.setExperimentId(new ExperimentIdentifier(experimentID));
+    creation.setParentIds(parentCodes.stream().map(DataSetPermId::new).collect(
+        Collectors.toList()));
+    creation.setTypeId(new EntityTypePermId("UNKNOWN", EntityKind.DATA_SET));
+
+    try
+    {
+      return openBIS.createUploadedDataSet(creation);
+    } catch (final Exception e)
+    {
+      e.printStackTrace();
+    }
+    return null;
+  }
+
+  public static void main(String[] args) throws IOException {
+    String as = "";
+    String dss = "";
+    String user = "";
+    String pass = "";
+
+    OpenBIS authentication =
+        new OpenBIS(as, dss);
+      authentication.login(user, pass);
+    OpenbisConnector c = new OpenbisConnector(authentication);
+
+    String space = "TEMP_PLAYGROUND";
+    String project = "TEMP_PLAYGROUND";
+    String experiment = "E123";
+    String expID = "/TEMP_PLAYGROUND/TEMP_PLAYGROUND/E123";
+    Path toUpload = Path.of("/Users/afriedrich/Downloads/cats");
+    //c.registerDataset(toUpload, expID, List.of("20240512182451335-425180", "20240512191248297-425181"));
+
+    String basePath = "/Users/afriedrich/Downloads/downloaded_cats";
+    //c.downloadDataset(basePath, "20240503175717814-263844");
+    c.listDatasetsOfExperiment(List.of(), experiment);
+
+  }
+
+
+private static void copyInputStreamToFile(InputStream inputStream, File file)
+    throws IOException {
+  try (FileOutputStream outputStream = new FileOutputStream(file, false)) {
+    int read;
+    byte[] bytes = new byte[8192];
+    while ((read = inputStream.read(bytes)) != -1) {
+      outputStream.write(bytes, 0, read);
+    }
+  }
+}
+
+public List<DataSet> listDatasetsOfExperiment(List<String> spaces, String experiment) {
+  DataSetSearchCriteria criteria = new DataSetSearchCriteria();
+  criteria.withExperiment().withCode().thatEquals(experiment);
+  if(!spaces.isEmpty()) {
+    criteria.withAndOperator();
+    criteria.withExperiment().withProject().withSpace().withCodes().thatIn(spaces);
+  }
+  DataSetFetchOptions options = new DataSetFetchOptions();
+  options.withType();
+  options.withRegistrator();
+  options.withExperiment().withProject().withSpace();
+  return openBIS.searchDataSets(criteria, options).getObjects();
+}
+
+  public void downloadDataset(String targetPath, String datasetID) throws IOException {
+    DataSetFileDownloadOptions options = new DataSetFileDownloadOptions();
+    IDataSetFileId fileToDownload = new DataSetFilePermId(new DataSetPermId(datasetID),
+        "");//TODO test this path
+
+    // Setting recursive flag to true will return both the subfolder directory object AND file3.txt
+    options.setRecursive(true);
+
+    // Setting recursive flag to false will return just the meta data of the directory object
+    //options.setRecursive(false);
```

Comment:
NIT: Remove this, ideally the method explanation is moved to the openbis method anyway. One solution could be to have an interface between the openbis methods and our methods for abstraction and descriptions like this  🤔